backend/Bindings/Edna.Bindings.LtiAdvantage/Services/NrpsClient.cs

```diff
@@ -111,5 +122,32 @@ public async Task<Member> GetById(string clientId, string tokenUrl, string audie
 
             return allMembers.FirstOrDefault(member => member.UserId.Equals(userId));
         }
+
+        private static string GetNextMembershipUrlFromHeaders(HttpResponseHeaders headers)
+        {
+            string linkUrls = headers.TryGetValues("Link", out var values) ? values.FirstOrDefault() : null;
+            if (string.IsNullOrEmpty(linkUrls))
+            {
+                return null;
+            }
+
+            string[] urls = linkUrls.Split(',');
+            string nextUrl = urls.FirstOrDefault(u => u.IndexOf("\"next\"", StringComparison.OrdinalIgnoreCase) >=0 );
+
+            if (string.IsNullOrEmpty(nextUrl))
+            {
+                return null;
+            }
+
+            // Regex to get the url from next url excluding < and >; characters.
+            Regex reg = new Regex("(?<=<)(.*?)(?=>;)");

```

Comment:
Seems like we would be creating this regex for every membership call. Is it possible to move this to a class variable so that the complete Regex object is not created for every call and can be reused across them?

Reply:
Just for own curiosity, the problem with the following regex is the fact that we might need to use groups to get the actual value, or something else?
```
Regex reg = new Regex("<(.*?)>; rel=next");
```

Reply:
yes, that regex would work as well but the current is also working just fine

Reply:
Thanks for validating this. The reason I placed this comment is that I've got confused with the look-ahead and that took me some time to figure out. Hence, was curious to understand the background on using lookahead in this particular case.